### PR TITLE
[M68K] Added basic groups

### PR DESCRIPTION
**Pull request by emoon**
Added support for basic groups in the M68K backend. Also did some minor cleanups/whitespace fixes while at it.

Relates to this issue https://github.com/aquynh/capstone/issues/494

### CODE REVIEW

**Comment by emoon** [comment]
cc @nplanel

File: arch/M68K/M68KDisassembler.c [hunk 1]
```diff
@@ -2018,7 +2026,7 @@ static void d68020_cpgen(m68k_info *info)
 		case 0x28: MCInst_setOpcode(info->inst, M68K_INS_FSUB); supports_single_op = false; break;
 		case 0x38: MCInst_setOpcode(info->inst, M68K_INS_FCMP); supports_single_op = false; break;
 		case 0x3a: MCInst_setOpcode(info->inst, M68K_INS_FTST); break;
-		default:	
+		default:
 				   break;
```
Comment (aquynh) [inline arch/M68K/M68KDisassembler.c]: looks like this "break" does not have proper indentation?

File: arch/M68K/M68KDisassembler.c [hunk 2]
```diff
@@ -2339,12 +2347,14 @@ static void d68020_extb_32(m68k_info *info)
 
 static void d68000_jmp(m68k_info *info)
 {
+	set_insn_group(info, M68K_GRP_JUMP);
 	cs_m68k* ext = build_init_op(info, M68K_INS_JMP, 1, 0);
```
Comment (aquynh) [inline arch/M68K/M68KDisassembler.c]: we prefer c89 style. can you move the declaration of `ext` to the beginning of function, before `set_insn_group()` ?

we have this same problem in other places.

File: arch/M68K/M68KDisassembler.h [hunk 3]
```diff
@@ -18,6 +18,8 @@ typedef struct m68k_info {
 	unsigned int type;
 	unsigned int address_mask; /* Address mask to simulate address lines */
 	cs_m68k extension;
+	uint8_t groups[8];
+	uint8_t groups_count;
```
Comment (aquynh) [inline arch/M68K/M68KDisassembler.h]: self noted: this requires change on Python bindings. if you can fix Python binding, that will be awesome.

File: arch/M68K/M68KDisassembler.c (hunk 2 above)
Comment (emoon) [inline arch/M68K/M68KDisassembler.c]: Alright. I will fix. Maybe having c89 flags for the compilers to enforce this then?

File: arch/M68K/M68KDisassembler.h (hunk 3 above)
Comment (emoon) [inline arch/M68K/M68KDisassembler.h]: Are you sure? This is an internal data structure.

File: include/capstone/m68k.h [hunk 4]
```diff
@@ -78,7 +78,7 @@ typedef enum m68k_reg {
 
 //> M68K Addressing Modes
 typedef enum m68k_adress_mode {
-	M68K_AM_NONE = 0,			// No address mode. 
+	M68K_AM_NONE = 0,			// No address mode.
```
Comment (aquynh) [inline include/capstone/m68k.h]: i didnt pay attention to this before. for consistency, would you mind renaming this to M68K_AM_INVALID?

File: include/capstone/m68k.h [hunk 5]
```diff
@@ -569,6 +569,16 @@ typedef enum m68k_insn {
 	M68K_INS_UNPK,
 } m68k_insn;
 
+//> Group of M68K instructions
+typedef enum m68k_group_type {
+	M68K_GRP_INVALID = 0,  // CS_GRUP_INVALID
+	M68K_GRP_JUMP,  // = CS_GRP_JUMP
+	M68K_GRP_RET = 3,  // = CS_GRP_RET
+	M68K_GRP_IRET, // = CS_GRP_IRET
```
Comment (aquynh) [inline include/capstone/m68k.h]: this should be `M68K_GRP_IRET = 5` because `CS_GRP_IRET = 5`

**Comment by aquynh** [comment]
this looks good except some minor issues. if @nplanel has no objection, i will merge this.

thanks

File: arch/M68K/M68KDisassembler.h (hunk 3 above)
Comment (aquynh) [inline arch/M68K/M68KDisassembler.h]: oh yes it seems so. then would you mind extending `test_m68.py` to handle these instruction groups, so it does exactly what `test_m68.c` does?

File: arch/M68K/M68KDisassembler.c (hunk 2 above)
Comment (aquynh) [inline arch/M68K/M68KDisassembler.c]: yes, if you can add c89 flags to Makefile, that will be nice.

File: include/capstone/m68k.h (hunk 4 above)
Comment (emoon) [inline include/capstone/m68k.h]: Can I do it in another PR? Otherwise this PR does more than it "should"

Comment (aquynh) [inline include/capstone/m68k.h]: sure, thanks.

**Comment by emoon** [comment]
Fixed some of the minor issues (except ```M68K_AM_INVALID```)

File: bindings/python/test_m68k.py [hunk 6]
```diff
@@ -40,12 +40,14 @@
...
 	17: "Absolute Data Addressing  - Long",
 	18: "Immidate value",
-}    
+}
 
 def print_insn_detail(insn):
     if len(insn.operands) > 0:
         print("\top_count: %u" % (len(insn.operands)))
 
+	print("\tgroups_count: %u" % len(insn.groups))
```
Comment (nplanel) [inline bindings/python/test_m68k.py]: @emoon 
This will cause an exception as insn.groups didn't exist

@aquynh, @emoon :
Only public cs_m68k structure is binded to python. (like other arch)
"detail->groups_count" is private.

In other arch, that's the same behavior on x86, you can't access to insn->detail, insn->detail->group in python

May groups field should move to cs_m68k ... an others arch ?

Comment (emoon) [inline bindings/python/test_m68k.py]: Hum... the groups_counts work when I test this in python and prints the correct value.

Comment (nplanel) [inline bindings/python/test_m68k.py]: @emoon : My bad, I apologize :( .... forgot to check in capstone/__init__.py  ; python binding have the magic trick implemented for all arch.

BTW : You should change the tabs to space in front of the print, or python will report : 
'''
  File "test_m68k.py", line 48
    print("\tgroups_count: %u" % len(insn.groups))
                                                 ^
TabError: inconsistent use of tabs and spaces in indentation
'''

Comment (emoon) [inline bindings/python/test_m68k.py]: Thanks for the PR. Would it be fine to submit it directly to the "next" branch directly or should i merge it within this PR?

**Comment by aquynh** [comment]
yes, you can push that commit in the same branch, and i will merge them all.

thanks

**Comment by emoon** [comment]
Alright. I will look at it when I get home.

**Comment by aquynh** [comment]
merged, thanks!

**Comment by emoon** [comment]
Sweet!
